Symfony/CS/AbstractFixer.php

```diff
@@ -21,22 +21,9 @@
     /**
      * {@inheritdoc}
      */
-    public function getLevel()
+    public function configure(array $configuration = null)
     {
-        static $map = array(
-            'PSR1' => FixerInterface::PSR1_LEVEL,
-            'PSR2' => FixerInterface::PSR2_LEVEL,
-            'Symfony' => FixerInterface::SYMFONY_LEVEL,
-            'Contrib' => FixerInterface::CONTRIB_LEVEL,
-        );
-
-        $level = current(explode('\\', substr(get_called_class(), strlen(__NAMESPACE__.'\\Fixer\\'))));
-
-        if (!isset($map[$level])) {
-            throw new \LogicException(sprintf('Can not determine Fixer level: "%s".', $level));
-        }
-
-        return $map[$level];
+        return;
```

Comment:
> This PR may be big, so it may be worth to read changes commit by commit instead of all at once.
> Please split discussion into two parts - first do we like changes described below and implemented, and then, if we agreed about changes, I will fill the docs and review of the code itself may begin.

;)